Templates/make_jenkins_EC2-Agent-Windows-instance.tmplt.json

```diff
@@ -0,0 +1,1613 @@
+        },
+        "AppScriptParams": {
+            "Description": "Parameter string to pass to the application script. Ignored if \"AppScriptUrl\" is blank",
+            "Type": "String"
+        },
+        "AppScriptUrl": {
+            "AllowedPattern": "^$|^s3://(.*)/(.*)\\.(bat|ps1)$",
+            "ConstraintDescription": "Must use an S3 URL (starts with \"s3://\") to a .bat or .ps1 file",
+            "Default": "",
+            "Description": "(Optional) S3 URL to the .ps1 or .bat application script in an S3 bucket (s3://). Leave blank to launch without an application script. If specified, an appropriate \"InstanceRole\" is required",
+            "Type": "String"
+        },
+        "AppVolumeDevice": {
+            "AllowedValues": [
+                "",
+                "xvdf",
+                "xvde",
+                "xvdg",
+                "xvdh",
+                "xvdi"
+            ],
+            "Default": "",
+            "Description": "(Optional) Device to mount an extra EBS volume. Leave blank to launch without an extra application volume",
+            "Type": "String"
+        },
+        "AppVolumeSize": {
+            "ConstraintDescription": "Must be between 1GB and 16384GB.",
+            "Default": "1",
+            "Description": "Size in GB of the EBS volume to create. Ignored if \"AppVolumeDevice\" is blank",
+            "MaxValue": "16384",
+            "MinValue": "1",
+            "Type": "Number"
+        },
+        "AppVolumeType": {
+            "AllowedValues": [
+                "gp2",
+                "io1",
+                "sc1",
+                "st1",
+                "standard"
+            ],
+            "Default": "gp2",
+            "Description": "Type of EBS volume to create. Ignored if \"AppVolumeDevice\" is blank",
+            "Type": "String"
+        },
+        "CfnEndpointUrl": {
+            "AllowedPattern": "^$|^http[s]?://.*$",
+            "Default": "https://cloudformation.us-east-1.amazonaws.com",
+            "Description": "(Optional) URL to the CloudFormation Endpoint. e.g. https://cloudformation.us-east-1.amazonaws.com",
+            "Type": "String"
+        },
+        "ChainScriptUrl": {
+          "AllowedPattern": "^$|^http[s]?://.*$|s3://.*$",
+          "Default": "",
+          "Description": "(Optional) URL to a \"chain\" script - a script to install add-ons not part of the core agent install.",
+          "Type": "String"
+        }, 
+        "ChainScriptParams": {
+          "Default": "",
+          "Description": "(Optional) Params to pass to chain load.",
+          "Type": "String"
+        }, 
+        "CloudWatchAgentUrl": {
+            "AllowedPattern": "^$|^s3://.*$",
+            "Default": "",
+            "Description": "(Optional) S3 URL to CloudWatch Agent installer. Example: s3://amazoncloudwatch-agent/windows/amd64/latest/AmazonCloudWatchAgent.zip",
+            "Type": "String"
+        },
+        "InstanceRole": {
```

Comment:
Might want to validate the need for this: attaching Instance-roles to Jenkins agents can be *VERY* dangerous. Attaching instance-roles (absent further restrictions installed by way of your app-script) gives _every_  job run on that agent access to any privileges granted to the agent-node. This can be seriously-bad juju. 

If you're looking to preserve flexibility by allowing attaching of instance-roles, you probably want to update the parameter's `Description` to add some kind of "DANGER WILL ROBINSON" kind of notation.